cmd/root_test.go

```diff
@@ -85,42 +88,63 @@ func TestRootCmd_OutputFlagTextVsJSON(t *testing.T) {
 	testutils.CheckExpectedError(t, err, nil)
 
-	err = testutils_cobra.ExecutePingcli(t, "--no-color=false")
+	err = testutils_cobra.ExecutePingcli(t, "--"+options.RootColorOption.CobraParamName+"=false")
 	testutils.CheckExpectedError(t, err, nil)
 }
 
 // Test Root Command fails when provided an invalid value for the --no-color flag
 func TestRootCmd_InvalidColorFlag(t *testing.T) {
 	expectedErrorPattern := `^invalid argument "invalid" for ".*" flag: strconv\.ParseBool: parsing "invalid": invalid syntax$`
-	err := testutils_cobra.ExecutePingcli(t, "--no-color=invalid")
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootColorOption.CobraParamName+"=invalid")
 	testutils.CheckExpectedError(t, err, &expectedErrorPattern)
 }
 
 // Test Root Command Executes when provided the --config flag
 func TestRootCmd_ConfigFlag(t *testing.T) {
-	err := testutils_cobra.ExecutePingcli(t, "--config", "config.yaml")
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootConfigOption.CobraParamName, "config.yaml")
 	testutils.CheckExpectedError(t, err, nil)
 }
 
 // Test Root Command fails when provided no value for the --config flag
 func TestRootCmd_NoValueConfigFlag(t *testing.T) {
 	expectedErrorPattern := `^flag needs an argument: --config$`
-	err := testutils_cobra.ExecutePingcli(t, "--config")
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootConfigOption.CobraParamName)
 	testutils.CheckExpectedError(t, err, &expectedErrorPattern)
 }
 
 // Test Root Command Executes when provided the --profile flag
 func TestRootCmd_ProfileFlag(t *testing.T) {
-	err := testutils_cobra.ExecutePingcli(t, "--profile", "default")
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootProfileOption.CobraParamName, "default")
 	testutils.CheckExpectedError(t, err, nil)
 }
 
 // Test Root Command fails when provided no value for the --profile flag
 func TestRootCmd_NoValueProfileFlag(t *testing.T) {
 	expectedErrorPattern := `^flag needs an argument: --profile$`
-	err := testutils_cobra.ExecutePingcli(t, "--profile")
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootProfileOption.CobraParamName)
 	testutils.CheckExpectedError(t, err, &expectedErrorPattern)
 }
+
+// // Test Root Command Detailed Exit Code Flag
+func TestRootCmd_DetailedExitCodeFlag(t *testing.T) {
+	err := testutils_cobra.ExecutePingcli(t, "--"+options.RootDetailedExitCodeOption.CobraParamName)
+	testutils.CheckExpectedError(t, err, nil)
+
+	err = testutils_cobra.ExecutePingcli(t, "-"+options.RootDetailedExitCodeOption.Flag.Shorthand)
+	testutils.CheckExpectedError(t, err, nil)
+}
+
+// // Test Root Command Detailed Exit Code Flag with output Warn
+func TestRootCmd_DetailedExitCodeWarnLoggedFunc(t *testing.T) {
+	testutils_viper.InitVipers(t)
+	t.Setenv(options.RootDetailedExitCodeOption.EnvVar, "true")
+	output.Warn("test warning", nil)
+
+	warnLogged, err := output.DetailedExitCodeWarnLogged()
```

Comment:
I think we want the logic reordered here

1. Check expected error is nil
2. Check warnLogged is true, since we did log a warning above